sql/updates/pre-update.sql

```diff
@@ -27,3 +27,24 @@ AS '@LOADER_PATHNAME@', 'ts_bgw_db_workers_restart'
 LANGUAGE C VOLATILE;
 
 SELECT _timescaledb_internal.restart_background_workers();
+
+-- Table for ACL and initprivs of tables. 
+CREATE TABLE saved_privs(
+       tmpnsp name,
+       tmpname name,
+       tmpacl aclitem[],
+       tmpini aclitem[],
+       UNIQUE (tmpnsp, tmpname));
+
+-- We save away both the ACL and the initprivs for all tables in the
+-- extension (but not for chunks and internal objects) so that we can
+-- restore them to the proper state after the update.
+INSERT INTO saved_privs
+SELECT nspname, relname, relacl, initprivs
+  FROM pg_class cl JOIN pg_namespace ns ON ns.oid = relnamespace
+                   JOIN pg_init_privs ip ON ip.objoid = cl.oid
+WHERE nspname IN ('_timescaledb_catalog', '_timescaledb_config')
+   OR nspname = '_timescaledb_internal'
+  AND relname IN ('hypertable_chunk_local_size', 'compressed_chunk_stats');
```

Comment:
If we do this for _timescaledb_internal, it should cover all the objects that we create in that schema during extension installation

Reply:
Since we would have to list the objects from _timescaledb_internal  here, it would be worth adding a comment in the README that pre/post and need to be updated for any new catalog objects.

Reply:
We need to have a better solution for how to handle and differentiate between objects that are internal to timescaledb objects and objects that we use as part of the implementation. A separate schema would be the easiest solution, but I am reluctant to add this change to this PR and intend to take it separately. For now, I just handle the internal views and tables.